router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming.go

```diff
@@ -382,67 +384,50 @@ func (m *Manager) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	var anyoneInProgress bool
-	for _, info := range infos {
-		inProgress, err := m.pollForImportInfo(ctx, info)
-		if err != nil {
-			info.Failed = true
-			info.Reason = err.Error()
-			continue
-		}
-		anyoneInProgress = anyoneInProgress || inProgress
-	}
-	if anyoneInProgress {
+	if anyInProgress := m.processPollImportInfos(ctx, importInfos); anyInProgress {
+		m.stats.pollingInProgress.Increment()
 		return common.PollStatusResponse{InProgress: true}
 	}
 
-	failedInfos := lo.Filter(infos, func(info *importInfo, index int) bool {
+	updatedImportInfos := lo.Map(importInfos, func(item *importInfo, index int) *importInfo {
+		return m.polledImportInfoMap[item.ChannelID]
+	})
+	failedImports := lo.Filter(updatedImportInfos, func(info *importInfo, index int) bool {
 		return info.Failed
 	})
-	for _, info := range failedInfos {
-		m.logger.Warnn("Failed to poll channel offset",
-			logger.NewStringField("channelId", info.ChannelID),
-			logger.NewStringField("offset", info.Offset),
-			logger.NewStringField("table", info.Table),
-			logger.NewStringField("reason", info.Reason),
-		)
+	m.cleanupFailedImports(ctx, failedImports)
+	m.updateJobStatistics(updatedImportInfos)
+	m.polledImportInfoMap = make(map[string]*importInfo)
 
-		if deleteErr := m.deleteChannel(ctx, info.Table, info.ChannelID); deleteErr != nil {
-			m.logger.Warnn("Failed to delete channel",
-				logger.NewStringField("channelId", info.ChannelID),
-				logger.NewStringField("table", info.Table),
-				obskit.Error(deleteErr),
-			)
+	return m.buildPollStatusResponse(updatedImportInfos, failedImports)
+}
+
+func (m *Manager) processPollImportInfos(ctx context.Context, infos []*importInfo) bool {
+	var anyInProgress bool
+	for i := range infos {
+		info := infos[i]
+
+		if _, alreadyProcessed := m.polledImportInfoMap[info.ChannelID]; alreadyProcessed {
+			continue
 		}
-	}
 
-	var successJobsCount, failedJobsCount int
-	for _, info := range infos {
-		if info.Failed {
-			failedJobsCount += info.Count
-		} else {
-			successJobsCount += info.Count
+		inProgress, err := m.getImportStatus(ctx, info)
+		if err != nil {
+			info.Failed = true
+			info.Reason = err.Error()
+			m.polledImportInfoMap[info.ChannelID] = info
```

Comment:
If the status API fails which means that the channel got invalidated/corrupted. No point in doing the poll again because we will get the same result.

Reply:
Can status api can also fail due to some network issue or some other issue, do we want to classify those failures separately?

Reply:
Yes. Anything can fail at any time in general right in Software? 

1. We already have a retry policy on the APIs we make to the Java service. 
2. It would be better to mark them as `failed`, instead of `retrying`/`waiting` indefinitely. Waiting indefinitely would cause other events to get blocked till the Polling is completed.